python/PyQt6/core/auto_additions/qgis.py

```diff
@@ -1,42 +1,56 @@
 # The following has been generated automatically from src/core/qgis.h
+Qgis.Info = Qgis.MessageLevel.Info
+Qgis.Warning = Qgis.MessageLevel.Warning
+Qgis.Critical = Qgis.MessageLevel.Critical
+Qgis.Success = Qgis.MessageLevel.Success
+Qgis.NoLevel = Qgis.MessageLevel.NoLevel
 Qgis.MessageLevel.baseClass = Qgis
 QgsMapLayer.LayerType = Qgis.LayerType
 # monkey patching scoped based enum
 QgsMapLayer.VectorLayer = Qgis.LayerType.Vector
 QgsMapLayer.LayerType.VectorLayer = Qgis.LayerType.Vector
 QgsMapLayer.VectorLayer.is_monkey_patched = True
 QgsMapLayer.VectorLayer.__doc__ = "Vector layer"
+Qgis.Vector = Qgis.LayerType.Vector
```

Comment:
These lines aren't actually required -- we never supported `Qgis.Vector` in pyqgis. (Same with all the other similar examples in this file)

Reply:
Indeed, `Qgis.Vector`doesn't exist because we already mock for old enum `QgsMapLayer, LayerType` thanks to SIP_MONKEYPATCH_SCOPEENUM_UNNEST

But, if I understand correctly:
- Qgis.Info exists because it is a classic enum
- Qgis.SqlQuery doesn't exist because it's an enum but inherits from int
- Qgis.UnknownDataType exists because it has SIP_MONKEYPATCH_SCOPEENUM_UNNEST mocked with Qgis.DataType

Is there another logic I missing here ? It doesn't seems very obvious for API user whether you should use an enum scoped or unscoped

Could be not mock all of them and set deprecation on warning on unscoped enum (if we do want to get rid of all of them)?

Reply:
> Indeed, Qgis.Vectordoesn't exist because we already mock for old enum QgsMapLayer, LayerType thanks to SIP_MONKEYPATCH_SCOPEENUM_UNNEST

> But, if I understand correctly:

> Qgis.Info exists because it is a classic enum

Correct

> Qgis.SqlQuery doesn't exist because it's an enum but inherits from int

Actually it doesn't because it's an enum class, and never was a non class enum (hence no need for backward compatibility)

> Qgis.UnknownDataType exists because it has SIP_MONKEYPATCH_SCOPEENUM_UNNEST mocked with Qgis.DataType

Correct -- it was upgraded to an enum class at some stage, hence the need for the old backward compatibility logic.

But this one is a bit on an exception -- usually SIP_MONKEYPATCH_SCOPEENUM_UNNEST is used to specify a **different** class, in which case there's no need to insert the values directly into the Qgis class. 

So eg GeometryType is patched for QgsWkbTypes, and we shouldn't have Qgis.Line or Qgis.LineGeometry.

> Is there another logic I missing here ? It doesn't seems very obvious for API user whether you should use an enum scoped or unscoped

From a Pyqgis point of view you could always refer to enums using the enum name -- eg Qgis.MessageLevel.Info works, even though that's not an enum class. So best practice would be to always do this, and then there's no inconsistency/porting issues.

> Could be not mock all of them and set deprecation on warning on unscoped enum (if we do want to get rid of all of them)?

I think that's risky, there's issues if the same value is used for different enums. Eg Line is present for SymbolType, GeometryType, BrowserLayerType, LabelPlacement, ...

So the logic in sipify should be:

-if enum class and monkey patched, values should be accessible direct from the old class. 
- if enum class and monkey patched, values should not be directly accessible from the new class (unless the new and old class are the same)
- if enum class and not monkey patched, values are only accessible using their full class.enum.value form
- if NOT enum class, values should be directly accessible from the class OR by explicitly stating the enum name

Reply:
Thank you for detailed explanation of the logic, I'm gonna make the needed modifications.

> From a Pyqgis point of view you could always refer to enums using the enum name -- eg Qgis.MessageLevel.Info works, even though that's not an enum class. So best practice would be to always do this, and then there's no inconsistency/porting issues.

I would in favor of enforcing scoped enum by deprecation warning but obviously keeping the retrocompatibility with the way it is now.

Reply:
Getting a deprecation warning when using the non scoped versions would be fantastic! Do you think that's possible?

Reply:
> I'm gonna make the needed modifications.

Done. Actually, I just had to [limit]( https://github.com/qgis/QGIS/pull/55568/commits/ddcd424ebf9383829d55bc5b95ad4d14646af622#diff-4550c83551335b5e82f1649f7ca356831a1feb5afadedb64c8b98f47002eb384L1226) the mock for PyQt6 for non class enum

I also reverted some sip file modifications that are related for later PR, sorry about that. Everything is now according to sipify.pl state.

> Getting a deprecation warning when using the non scoped versions would be fantastic! Do you think that's possible?

I was thinking of [that](https://dev.to/hckjck/python-deprecation-2mof). We could generate those with sipify.pl